### PR TITLE
fix: null index causes NPE in field action service

### PR DESCRIPTION
**Pull request by igarashitm**
Fixes: #3329
Implicit unboxing causes NPE

### CODE REVIEW

File: lib/core/src/main/java/io/atlasmap/core/DefaultAtlasFieldActionService.java [hunk 1]
```diff
@@ -799,7 +799,7 @@ private void extractFlatListValuesFromFieldGroup(AtlasInternalSession session,
                 // Collection field
                 List<Field> fields = ((FieldGroup) subField).getField();
                 if (fields != null && !fields.isEmpty()) {
-                    if (fieldGroup.getField().size() == 1 && (index == 0 || index == null)) {
+                    if (fieldGroup.getField().size() == 1 && (index == null || index == 0)) {
```
Comment (pleacu) [inline lib/core/src/main/java/io/atlasmap/core/DefaultAtlasFieldActionService.java:802]: Just trying to grok this - testing for 0 or null with the parenthetical expression shouldn't matter - how did this make a difference?

Comment (igarashitm) [reply lib/core/src/main/java/io/atlasmap/core/DefaultAtlasFieldActionService.java:802]: So if index is null, the former tries to "unbox" `Integer` into primitive `int` to evaluate `index == 0`, causes NPE.

Comment (pleacu) [reply lib/core/src/main/java/io/atlasmap/core/DefaultAtlasFieldActionService.java:802]: ah - sneaky - good catch...

**Review by pleacu — APPROVED** [review]
+1

**Pull request merged**